### PR TITLE
Schema resource added in REST API Spec for api-version 2021-04-01-preview

### PR DESCRIPTION
**Pull request by shinjiDev**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  December 2021
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  December 2021
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [X] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [X] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by solankisamir** [comment]
@shinjiDev can you fix the LintDiff and Spell Check errors in your PR

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimapis.json
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimapis.json:2146]: I don't see we have changed the `ApiSchemaController`

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimapis.json
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimapis.json:2207]: This requires change in the Backend also

**Review by solankisamir — CHANGES_REQUESTED** [review]
Let's postpone this to the next api-version. We are unnecessarily holding it hostage.

There are some decisions that need to be made
1. PUT should be long running
2. It should accept a Url Link instead of Inline Body, as ARM has 4MB limit.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimapis.json
Comment (maksimkim) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimapis.json:2146]: @solankisamir , feel free not to block merging 2021-04-01-preview into main on this PR. We'll request it separately.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimapis.json
Comment (shinjiDev) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimapis.json]: It was undo

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimapis.json
Comment (shinjiDev) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimapis.json]: @solankisamir It's going back to "schemas"

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimanagement.json
Comment (maksimkim) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimanagement.json:795]: what's the purpose of not allowing "managed"?

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimanagement.json
Comment (maksimkim) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimanagement.json:798]: why location is method?

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/examples/ApiManagementUpdateApiOperation.json
Comment (maksimkim) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/examples/ApiManagementUpdateApiOperation.json:65]: Why remove sample?

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimanagement.json
Comment (maksimkim) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimanagement.json:631]: Does it make sense to reuse SchemaIdParameter for both service level and api level schemas?

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimschema.json
Comment (maksimkim) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimschema.json:61]: Let's not allow to filter by anything other than "name"

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimschema.json
Comment (maksimkim) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimschema.json:157]: It's not going to produce anything but application/json. Schema type will be conveyed in schemaType property.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimschema.json
Comment (maksimkim) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimschema.json:246]: Also add "202"

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/examples/ApiManagementCreateSchema1.json
Comment (maksimkim) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/examples/ApiManagementCreateSchema1.json:11]: This seems to be an xml document. Could you add xml schema some simple xml schema instead?

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/examples/ApiManagementCreateSchema2.json
Comment (maksimkim) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/examples/ApiManagementCreateSchema2.json:11]: This seems to be a swagger document. Could you add some simple json schema instead?

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/examples/ApiManagementGetApiOperation.json
Comment (maksimkim) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/examples/ApiManagementGetApiOperation.json:28]: Why removing sample?

Comment (shinjiDev) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/examples/ApiManagementGetApiOperation.json:28]: Because RepresentationContract in definitions.json doesn't have "sample" field in this preview.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/examples/ApiManagementUpdateApiOperation.json
Comment (shinjiDev) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/examples/ApiManagementUpdateApiOperation.json:65]: Because RepresentationContract in definitions.json doesn't have "sample" field in this preview.

**Comment by chiragg4u** [comment]
Please go through the contribution checklist added as part of the review and mark them completed. It will help you fix most of the issues and improve the quality of swagger and help close the PR faster. This is a requirement before review can be continued.

**Comment by shinjiDev** [comment]
> Let's postpone this to the next api-version. We are unnecessarily holding it hostage.
> 
> There are some decisions that need to be made
> 
> 1. PUT should be long running
> 2. It should accept a Url Link instead of Inline Body, as ARM has 4MB limit.

We got breaking changes approval.

**Review by shinjiDev — COMMENTED** [review]
Thank you. It looks like everything is done.

**Review by solankisamir — APPROVED** [review]

**Comment by zhenglaizhang** [comment]
@fengzhou-msft could you help check if we can ignore the .net sdk failure? 
@ArcturusZhang could you help check and appove go SDK breaking change? 
@msyyc could you help check python SDK breaking change?

**Comment by maksimkim** [comment]
@zhenglaizhang is anything pending for this to be merged?

**Comment by zhenglaizhang** [comment]
> @zhenglaizhang is anything pending for this to be merged?

this PR is adding new API which requires ARM signed off per https://github.com/Azure/azure-rest-api-specs/pull/15667#issuecomment-899869862 others LGTM.

**Comment by shinjiDev** [comment]
> > @zhenglaizhang is anything pending for this to be merged?
> 
> this PR is adding new API which requires ARM signed off per [#15667 (comment)](https://github.com/Azure/azure-rest-api-specs/pull/15667#issuecomment-899869862) others LGTM.

checkbox added.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimschema.json
Comment (mentat9) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimschema.json:86]: >ErrorResponse [](http://example.com/codeflow?start=57&length=13)

You can use definitions from types.json below common-types for these standard ARM properties (e.g. ErrorResponse, Resource): "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse".

**Comment by mentat9** [comment]
Note: this PR is updating an existing API version rather than new service onboarding. Please update the submission form accordingly. I've reviewed the changes and left comments. Nothing is blocking for ARM signoff, but please consider the comments.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimschema.json
Comment (shinjiDev) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-04-01-preview/apimschema.json]: Done

**Comment by shinjiDev** [comment]
> Note: this PR is updating an existing API version rather than new service onboarding. Please update the submission form accordingly. I've reviewed the changes and left comments. Nothing is blocking for ARM signoff, but please consider the comments.

PR description updated.

**Pull request merged**
